manifests/opsmanager/install.pp

```diff
@@ -0,0 +1,48 @@
+# PRIVATE CLASS: do not call directly
+class mongodb::opsmanager::install {
+  #assert_private("You are calling a private class mongodb::opsmanager::install.")
+  $package_ensure = $mongodb::opsmanager::package_ensure
+  $package_name   = $mongodb::opsmanager::package_name
+  $download_url   = $mongodb::opsmanager::download_url
+
+  case $package_ensure {
+    true:     {
+      $my_package_ensure = 'present'
+      $file_ensure       = 'directory'
+    }
+    false:    {
+      $my_package_ensure = 'absent'
+      $file_ensure       = 'absent'
+    }
+    'absent': {
+      $my_package_ensure = 'absent'
+      $file_ensure       = 'absent'
+    }
+    'purged': {
+      $my_package_ensure = 'purged'
+      $file_ensure       = 'absent'
+    }
+    default:  {
+      $my_package_ensure = $package_ensure
+      $file_ensure       = 'present'
+    }
+  }
+
+  case $facts['os']['family'] {
+    'RedHat': {
+      $my_provider = 'rpm'
+    }
+    default: {
```

Comment:
Can we not put a `'Debian'` case in here?  I've just double checked, and package `provider` definitely defaults to `apt` on `osfamily == 'Debian'`.  Unlike `yum` (the default for RedHat), the `apt` provider doesn't have support for `source` and you need to use `dpkg` instead.